### PR TITLE
Feature/#2 signin landing and #50 auth

### PR DESCRIPTION
**Pull request by lily-law**
This PR completes the prototype Landing page

### CODE REVIEW

File: app/src/components/login/Login.tsx [hunk 1]
```diff
@@ -1,10 +1,112 @@
...
+    currentPassword.length < 8 && errors.push("at least 8 characters");
+    (!currentPassword.match(/[A-Z]/) || !currentPassword.match(/[a-z]/)) &&
+      errors.push("a mix of uppercase and lowercase");
+    setPasswordErrorMessage(
+      errors.length > 0
+        ? ["Password requires ", errors.join(" and ")].join("")
+        : ""
+    );
+    return errors.length > 0;
+  };
```
Comment (davideastmond) [inline app/src/components/login/Login.tsx]: Nice! Do you think it's a good idea to factor this out into a utility method?

**Review by davideastmond — APPROVED** [review]

File: app/src/components/login/Login.tsx (hunk 1 above)
Comment (lily-law) [reply app/src/components/login/Login.tsx]: Yes good idea, we'll probably want to use it for the signup too

**Comment by lily-law** [comment]
I've changed the way input works so now it:
- takes setValue in place of onChange
- validationMessenger takes in a function that should validate the string and return any error messages
- errorMessage prop is now simply an override so the parent can still set an error where needed

This makes using the component neater as less boilerplate is required

**Review by davideastmond — APPROVED** [review]
🍦 Looks good!

File: app/src/components/login/Login.tsx [hunk 2]
```diff
@@ -1,10 +1,95 @@
...
+    const errors = !isEmailValid(email) || isPasswordValid(password);
+    if (!errors) {
+      let req;
+      let reqError;
+      try {
+        req = await axios({
+          method: "post",
+          url: "/auth/local",
+          data: {
+            email,
```
Comment (davideastmond) [inline app/src/components/login/Login.tsx:27]: It might be a good idea to `.toLowerCase()` emails sent in the request?

Comment (lily-law) [reply app/src/components/login/Login.tsx:27]: Maybe better to do it in the back-end? seeing as it's bad practise to trust input from the client anyway.

**Pull request merged**